src/WebJobs.Script/Rpc/LanguageWorkerChannel.cs

```diff
@@ -322,13 +323,25 @@ internal void PublishRpcChannelReadyEvent(RpcEvent initEvent)
             }
         }
 
-        public void RegisterFunctions(IObservable<FunctionMetadata> functionRegistrations)
+        public void SetupFunctionInvocationBuffers(IEnumerable<FunctionMetadata> functions)
         {
-            if (functionRegistrations == null)
+            _functions = functions;
+            foreach (FunctionMetadata metadata in functions)
             {
-                throw new ArgumentNullException(nameof(functionRegistrations));
+                _workerChannelLogger.LogDebug("Setting up FunctionInvocationBuffer for function:{functionName} with functionId:{id}", metadata.Name, metadata.FunctionId);
+                _functionInputBuffers[metadata.FunctionId] = new BufferBlock<ScriptInvocationContext>();
+            }
+        }
+
+        public void SendFunctionLoadRequests()
+        {
+            if (_functions != null)
+            {
+                foreach (FunctionMetadata metadata in _functions)
+                {
+                    SendFunctionLoadRequest(metadata);
```

Comment:
Why'd this change? You used to subscribe to the event...

Reply:
Removed it to simplify.  There is no need to subscribe as langauge worker gets initialized only after function indexing completes.